routing/router.go

```diff
@@ -1101,3 +1104,54 @@ func (r *ChannelRouter) AddProof(chanID lnwire.ShortChannelID,
 	info.AuthProof = proof
 	return r.cfg.Graph.UpdateChannelEdge(info)
 }
+
+// getGraphDiameter is a function that determines the maximum distance between
+// two nodes in the graph, where the distance is calculated as the minimum number
+// of hops from src to dest
+func (r *ChannelRouter) getGraphDiameter() (uint8,error) {
+	// local variable to represent the graph diameter
+	var diam uint8
+	// If Cache is non-empty, then return Cached value
+	if r.diamCache != 0 {
```

Comment:
Agreed! We would need to invalidate the cache when a new node is added, but im not sure where that would go. I plan on spending more time on it this weekend. Maybe we can have a call to discuss what your thoughts are.

The other option is to recompute every time, but that seems rediculous.

What we would need is a way to determine whether we need to update diamCache when a node is added without recomputing diamCache, e.g. if we kept a list of all src and dest nodes that are or length diamCache. If you attach a new node to one of those, or remove an a src/dest node (and there is only one such pair) we increment/ decrement diamCache and recompute the set.

Maybe we store all the paths in a maxHeap and use a hashtable to store a pointer to the node in the head